PixelPaint+LibFileSystemAccessClient: Port to `Core::Stream`

krkk opened the pull request.

krkk commented on Userland/Libraries/LibFileSystemAccessClient/Client.h:
```diff
@@ -18,8 +19,21 @@
 namespace FileSystemAccessClient {
 
+class Client;
+class File : public Core::Stream::File {
+public:
+    static ErrorOr<NonnullOwnPtr<File>> adopt_fd(Badge<Client>, String filename, int fd, Core::Stream::OpenMode);
+
+    String filename() const { return m_filename; }
+
+private:
+    File(String filename, Core::Stream::File);
+
+    String m_filename;
+};
+
 using DeprecatedResult = ErrorOr<NonnullRefPtr<Core::File>>;
-using Result = ErrorOr<NonnullOwnPtr<Core::Stream::File>>;
+using Result = ErrorOr<NonnullOwnPtr<File>>;
```
I'm not so sure about this one tbh. It feels a bit hackish, but it's probably better than having a result type like this
```cc
struct FileAccessResult {
    NonnullOwnPtr<Core::Stream::File> file;
    String filename;
};
using Result = ErrorOr<FileAccessResult>;
```

LucasChollet replied:
That's something that already have been discussed on discord:
https://discord.com/channels/830522505605283862/830739873119207426/1050376615361773588

krkk replied:
oh, thanks for the link. I will change the PR.

LucasChollet commented:
On your second commit, the title doesn't say that you are also implementing `request_file_read_only_approved`

AtkinsSJ commented:
This needs rebasing, but the PR is relied on is merged. :^)

LucasChollet commented on Userland/Libraries/LibFileSystemAccessClient/Client.h:
```diff
@@ -18,20 +19,41 @@
+    {
+    }
+
+    Core::Stream::File& stream() const { return *m_stream; }
```
The `const` here is quite confusing IMO, can't you return a `const&`?

krkk replied:
but that would make the most important calls to it (like `read()` and `write()`) inaccessible, right?

AtkinsSJ replied:
Right now, this code means "Someone with a const FSAC::File can get the Core::Stream::File from it and call arbitrary methods." That makes sense to me - a const Core::Stream::File would be useless because you can't read or write to it.

LucasChollet replied:
Why does it make sense to be able to read from a const FSAC::File but not from a const Stream::File?

LucasChollet reviewed (commented):
You can undraft it :^)

AtkinsSJ commented on Userland/Libraries/LibFileSystemAccessClient/Client.cpp:
```diff
@@ -48,6 +48,31 @@ DeprecatedResult Client::try_request_file_read_only_approved_deprecated(GUI::Win
     return handle_promise<DeprecatedResult>(id);
 }
 
+Result Client::request_file_read_only_approved(GUI::Window* parent_window, DeprecatedString const& path)
```
The commit title doesn't fit the standards for it being on one line. It's annoying with LibFileSystemAccessClient being so long, but maybe something like `LibFileSystemAccessClient: Add functions returning FSAC::File`?

AtkinsSJ reviewed (changes requested):
One little nit-pick, otherwise this is good!

krkk commented:
changes since push:
- rebased commits
- reworded first commit a bit (functions won't return Core::Stream directly, but will *use* it)
- changed title in the third commit as suggested
- changed buffer size from `BUFSIZ` (1024) to `PAGE_SIZE` (4096) in `PaletteWidget::load_palette_file()`.
  it looks like we don't really use bufsiz in our codebase, also it's more like a FILE* thing??? idunno.. ¯\\_(ツ)_/¯
